src/components/Clock.js

```diff
@@ -0,0 +1,47 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+export default class Clock extends Component {
+  state = {
+    date: new Date(),
+  }
+
+  componentDidMount() {
+    this.timer = setInterval(() => {
+      this.setState({
+        date: new Date(),
+      });
+
+      // eslint-disable-next-line
+      console.log(this.state.date.toLocaleTimeString())
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps) {
+    const prevName = prevProps.name;
+    const currName = this.props.name;
+
+    if (prevProps.name !== this.props.name) {
+      // eslint-disable-next-line
+      console.log(
+        `The Clock was renamed from ${prevName} to ${currName}`,
+      );
+    }
+  }
```

Comment:
It's great that you have used componentDidUpdate, but for this task, you should only console.log the new name only when clicking the button (so you could easily put the 'console.log to your 'renameClock' method)